src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -2798,3 +2801,239 @@ export function getMolecularProfileSamplesSet(
+            driverFilterTextElements.push('unknown');
+    }
+
+    const statusFilterTextElements: string[] = [];
+    if (statusFilterIsActive && isMutationType) {
+        geneFilterQuery.includeGermline &&
+            statusFilterTextElements.push('germline');
+        geneFilterQuery.includeSomatic &&
+            statusFilterTextElements.push('somatic');
+        geneFilterQuery.includeUnknownStatus &&
+            statusFilterTextElements.push('unknown');
+    }
+
+    const tierNames = tierFilterIsActive
+        ? _(geneFilterQuery.tiersBooleanMap)
+              .pickBy()
+              .keys()
+              .value()
+        : [];
+    if (tierFilterActive && geneFilterQuery.includeUnknownTier)
+        tierNames.push('unknown');
+
+    let driverFilterText = '';
+    if (driverFilterTextElements.length === 1)
+        driverFilterText = driverFilterTextElements[0];
+    else if (driverFilterTextElements.length > 1)
+        driverFilterText =
+            driverFilterTextElements.slice(0, -1).join(', ') +
+            ' or ' +
+            driverFilterTextElements.slice(-1);
+
+    let statusFilterText = '';
+    if (statusFilterTextElements.length === 1)
+        statusFilterText = statusFilterTextElements[0];
+    else if (statusFilterTextElements.length > 1)
+        statusFilterText =
+            statusFilterTextElements.slice(0, -1).join(', ') +
+            ' or ' +
+            statusFilterTextElements.slice(-1);
+
+    let tierFilterText = '';
+    if (tierNames.length === 1) tierFilterText = tierNames[0];
+    else if (tierNames.length > 1)
+        tierFilterText =
+            tierNames.slice(0, -1).join(', ') + ' or ' + tierNames.slice(-1);
+
+    return (
+        <div data-test={'groupedGeneFilterIcons'} className={styles.content}>
+            {driverFilterText && (
+                <div
+                    style={{
+                        display: 'flex',
+                        justifyContent: 'space-between',
+                    }}
+                >
+                    <span>annotation:</span>&nbsp;
+                    <span>{driverFilterText}</span>
+                </div>
+            )}
+            {statusFilterText && (
+                <div
+                    style={{
+                        display: 'flex',
+                        justifyContent: 'space-between',
+                    }}
+                >
+                    <span>status:</span>&nbsp;
+                    <span>{statusFilterText}</span>
+                </div>
+            )}
+            {tierFilterText && (
+                <div
+                    style={{
+                        display: 'flex',
+                        justifyContent: 'space-between',
+                    }}
+                >
+                    <span>category:</span>&nbsp;
+                    <span>{tierFilterText}</span>
+                </div>
+            )}
+        </div>
+    );
+}
+
+export function tierFilterActive(
```

Comment:
was this not covered by computed values in store?   

all of stuff in this file seems like it should be computed

Reply:
Yes, corrected in a earlier comment